specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2661 @@
+        "tags": [
+          "amlFilesystems"
+        ],
+        "description": "Create or update a amlFilesystem.",
+        "operationId": "amlFilesystems_CreateOrUpdate"
+      },
+      "patch": {
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "description": "Target resource group.",
+            "in": "path",
+            "name": "resourceGroupName",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "description": "Name for the amlFilesystem.  Alphanumerics, underscores, and hyphens. Start and end with alphanumeric.",
+            "in": "path",
+            "name": "amlFilesystemName",
+            "pattern": "^[0-9a-zA-Z][-0-9a-zA-Z_]{0,78}[0-9a-zA-Z]$",
+            "required": true,
+            "type": "string",
+            "minLength": 2,
+            "maxLength": 80
+          },
+          {
+            "description": "Object containing the user-selectable properties of the amlFilesystem. If read-only properties are included, they must match the existing values of those properties.",
+            "in": "body",
+            "name": "amlFilesystem",
+            "schema": {
+              "$ref": "#/definitions/amlFilesystem"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Updated the amlFilesystem.",
+            "schema": {
+              "$ref": "#/definitions/amlFilesystem"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "amlFilesystems_Update": {
+            "$ref": "./examples/amlFilesystems_Update.json"
+          }
+        },
+        "tags": [
+          "amlFilesystems"
+        ],
+        "description": "Update a amlFilesystem instance.",
+        "operationId": "amlFilesystems_Update"
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.StorageCache/amlFilesystems/{amlFilesystemName}/archive": {
+      "post": {
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "description": "Target resource group.",
+            "in": "path",
+            "name": "resourceGroupName",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
```

Comment:
Should the amlFilesystemName be defined below and $ref'ed (like SubscriptionIdParameter)?   We've cut/pasted it many times in the spec.

Reply:
I can try again.  I did this to the name and resource group and the autorest validator threw errors until I listed it out inline again.

Reply:
Fixed.  Needed -x-ms-parameter-location : method